cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/spec/PipelineSpecGenerator.java

```diff
@@ -132,8 +133,14 @@ protected <T extends PluginConfigurer & DatasetConfigurer> PipelineSpecGenerator
       Map<String, MacroEvaluator> evaluators = Collections.singletonMap(
           ConnectionMacroEvaluator.FUNCTION_NAME,
           new ConnectionMacroEvaluator(namespace, runtimeConfigurer));
+      Map<String, String> runtimeArgs = runtimeConfigurer.getRuntimeArguments();
```

Comment:
With the logic to pass in empty arguments above, I don't think we will need this logic here I believe?

Reply:
If this logic is not needed, you can put the argument in `SystemArguments`

Reply:
ack